### ISSUE TITLE
RTL bug may result in uncorrectable ECC error firing erroneously during mailbox command

### ISSUE THREAD

**Issue by calebofearth** [chipsalliance/caliptra-rtl#340]
Found this corner case in nightly regressions.

### Summary
Caliptra might get a spurious Mailbox SRAM ECC double-bit error interrupt in this case:
- mailbox command issued with dlen = Y (which is > X)
- ECC double-bit error occurs when writing to (X+1) in mailbox SRAM
- mailbox command completes, mbox goes idle
- NEW command arrives with dlen=X
- when uC reads data from mailbox, it reads up to X.
- Upon read from X, mbox reads from address X+1 to populate dataout (mbox_dataout reg update is blocked by mbox_rd_valid_f, as expected)
- Read from X+1 flags double-bit ECC error

### Impact
Firmware is expected to abort any command (with the failure status) for which an uncorrectable ECC error occurs. This scenario may cause firmware to abort a mailbox command for which no error occurred. Assuming SoC retries the command, the error won't cause any system failure/deadlock.

### Remediation
Solvable via firmware as-is. Firmware will simply abort the command, then should clear the mailbox contents to 0 before accepting a new command.

One proposed hardware fix would involve using the rd_valid signal to gate reads from mailbox SRAM. This is postponed until after 1.0.
